### PR TITLE
fix: add authenticatedpageroute so learner can create a shared record link

### PR DESCRIPTION
**Pull request by jsnwesson**
Description:

Resolves the bug APER-2606: [Learner unable to create public record without reauthenticating](https://2u-internal.atlassian.net/browse/APER-2603)

When the learner attempted to create a public record link, their clipboard copied the private URL and they weren't able to create a public URL unless they re-authenticated. This PR wraps the `ProgramRecordsList`, `ProgramCertificatesList`, and the private version of `ProgramRecord` with `AuthenticatedPageRoute`, which allows for the user to experience a redirect "to the login page when the route becomes active and the user is not authenticated" [source](https://openedx.github.io/frontend-platform/module-React.html#.AuthenticatedPageRoute). 

Note that the public view of `ProgramRecord` is not wrapped in the `AuthenticatedPageRoute`. This route needs to not be wrapped so that non-learners are able to access the public record via shared URL.

### CODE REVIEW

**Review by justinhynes — APPROVED** [review]

File: src/index.jsx [hunk 1]
```diff
@@ -29,7 +29,7 @@ subscribe(APP_READY, () => {
           <Routes>
             <Route
               path={ROUTES.PROGRAM_RECORDS}
-              element={<ProgramRecordsList />}
+              element={<AuthenticatedPageRoute><ProgramRecordsList /></AuthenticatedPageRoute>}
```
Comment (MaxFrank13) [inline src/index.jsx:32]: I was under the impressions that the `AuthenticatedPageRoute` component was actually a react-router `Route` of its own. This is indicated [in the docs](https://github.com/openedx/frontend-platform/blob/93ea5b25423e8b607fb1c8633778503d76f47509/src/react/AuthenticatedPageRoute.jsx#L9-L12), HOWEVER, the implementation used in `frontend-platform` test files just [uses it as a wrapper](https://github.com/openedx/frontend-platform/blob/93ea5b25423e8b607fb1c8633778503d76f47509/src/react/AuthenticatedPageRoute.test.jsx#L44) (like this PR does).

When digging into the code, I don't see anywhere that `AuthenticatedPageRoute` would ever return an actual `Route` component. See the [code in frontend-platform](https://github.com/openedx/frontend-platform/blob/93ea5b25423e8b607fb1c8633778503d76f47509/src/react/AuthenticatedPageRoute.jsx#L35-L37).

What's strange is that there ARE implementations where MFEs are adding props like `path` and `component` to the `AuthenticatedPageRoute` component. See [here in frontend-app-learner-portal-enterprise](https://github.com/openedx/frontend-app-learner-portal-enterprise/blob/451b94b85eeef96e7527968ecfe383fab58e71ff/src/components/app/App.jsx#L55-L56).

All of this to say, this PR is clearly fine but I'm also clearly a little confused so will be bringing this to fedX group to hopefully get some clarification on how this component is intended to be used.

Comment (jsnwesson) [reply src/index.jsx:32]: Hey thanks for bringing in all of these PR examples! Very valid point in terms of implementation. Originally I did try replacing the `Route` element with a `AuthenticatedPageRoute`, but then an error appeared: 

`[AuthenticatedPageRoute] is not a <Route> component. All component children of <Routes> must be a <Route> or <React.Fragment>`

This must be a hard requirement for frontend-platform v5, and previous PRs might need to be updated to reflect this rule.

Comment (jsnwesson) [reply src/index.jsx:32]: One other thing to note is that the comments[ in the docs](https://github.com/openedx/frontend-platform/blob/93ea5b25423e8b607fb1c8633778503d76f47509/src/react/AuthenticatedPageRoute.jsx#L9-L12) is from 4 years ago (found via git blame), so it should be updated to better represent its implementation.

Comment (jsnwesson) [reply src/index.jsx:32]: And lastly, it looks like the implementation in `frontend-app-learner-portal-enterprise` has placed the routes directly into `App.jsx` (and not wrapped with a `Routes` element), and this was done[ over two years ago](https://github.com/openedx/frontend-app-learner-portal-enterprise/blame/9b70f62e2df63754bd8c592c80e8498b6df4eb09/src/components/app/App.jsx#L30). And they're still using `frontend-platform v4`.

So from that, I think that `AuthenticatedPageRoute` once had served as a substitute for `Route`, but it no longer does in v5.

**Pull request merged**